doc: Remove documentation for child/parent image (#20005)
nrfconnect/sdk-nrf: samples/tfm/provisioning_image_net_core/README.rst

Running this provisioning image in the network core will disable the debugging access, as required by the Trusted Firmware-M (TF-M) provisioning process.
The debugging is disabled by enabling the APPROTECT in the UICR register.

> This sample is included by default as a image of the :ref:`provisioning image<provisioning_image>` sample which runs on the application core.

Reviewer: ```suggestion
This sample is included by default as an image of the :ref:`provisioning image<provisioning_image>` sample that runs on the application core.
```
